release/pkg/bundles/nutanix.go

```diff
@@ -0,0 +1,105 @@
+
+	var sourceBranch string
+	var componentChecksum string
+	bundleImageArtifacts := map[string]anywherev1alpha1.Image{}
+	bundleManifestArtifacts := map[string]anywherev1alpha1.Manifest{}
+	artifactHashes := []string{}
+
+	for _, componentName := range sortedComponentNames {
+		for _, artifact := range nutanixBundleArtifacts[componentName] {
+			if artifact.Image != nil {
+				imageArtifact := artifact.Image
+				if componentName == "cluster-api-provider-nutanix" {
+					sourceBranch = imageArtifact.SourcedFromBranch
+				}
+				bundleImageArtifact := anywherev1alpha1.Image{
+					Name:        imageArtifact.AssetName,
+					Description: fmt.Sprintf("Container image for %s image", imageArtifact.AssetName),
+					OS:          imageArtifact.OS,
+					Arch:        imageArtifact.Arch,
+					URI:         imageArtifact.ReleaseImageURI,
+					ImageDigest: imageDigests[imageArtifact.ReleaseImageURI],
+				}
+				bundleImageArtifacts[imageArtifact.AssetName] = bundleImageArtifact
+				artifactHashes = append(artifactHashes, bundleImageArtifact.ImageDigest)
+			}
+
+			if artifact.Manifest != nil {
+				manifestArtifact := artifact.Manifest
+				bundleManifestArtifact := anywherev1alpha1.Manifest{
+					URI: manifestArtifact.ReleaseCdnURI,
+				}
+
+				bundleManifestArtifacts[manifestArtifact.ReleaseName] = bundleManifestArtifact
+
+				manifestHash, err := version.GenerateManifestHash(r, manifestArtifact)
+				if err != nil {
+					return anywherev1alpha1.NutanixBundle{}, err
+				}
+				artifactHashes = append(artifactHashes, manifestHash)
+			}
+		}
+	}
+
+	nutanixCloudProviderArtifacts := r.BundleArtifactsTable[fmt.Sprintf("cloud-provider-nutanix-%s", eksDReleaseChannel)]
+
+	for _, artifact := range nutanixCloudProviderArtifacts {
+		imageArtifact := artifact.Image
+
+		bundleArtifact := anywherev1alpha1.Image{
+			Name:        imageArtifact.AssetName,
+			Description: fmt.Sprintf("Container image for %s image", imageArtifact.AssetName),
+			OS:          imageArtifact.OS,
+			Arch:        imageArtifact.Arch,
+			URI:         imageArtifact.ReleaseImageURI,
+			ImageDigest: imageDigests[imageArtifact.ReleaseImageURI],
+		}
+		bundleImageArtifacts[imageArtifact.AssetName] = bundleArtifact
+		artifactHashes = append(artifactHashes, bundleArtifact.ImageDigest)
+	}
+
+	if r.DryRun {
+		componentChecksum = version.FakeComponentChecksum
+	} else {
+		componentChecksum = version.GenerateComponentHash(artifactHashes, r.DryRun)
+	}
+	version, err := version.BuildComponentVersion(
+		version.NewVersionerWithGITTAG(r.BuildRepoSource, constants.CapxProjectPath, sourceBranch, r),
+		componentChecksum,
+	)
+	if err != nil {
+		return anywherev1alpha1.NutanixBundle{}, errors.Wrapf(err, "Error getting version for cluster-api-provider-nutanix")
+	}
+
+	bundle := anywherev1alpha1.NutanixBundle{
+		Version:              version,
+		ClusterAPIController: bundleImageArtifacts["cluster-api-nutanix-controller"],
```

Comment:
This should match the `imageArtifact.AssetName` set in config/bundle_release.go since that's what is being used as the map key above.
```suggestion
		ClusterAPIController: bundleImageArtifacts["cluster-api-provider-nutanix"],
```